invenio_swh/service.py

```diff
@@ -103,14 +103,9 @@ def sync_status(self, id_, uow=None):
         deposit = deposit_res.deposit
         if not deposit:
             return
-        if deposit.status == SWHDepositStatus.FAILED:
```

Comment:
Removed the check from `sync_status`, otherwise we can't "recover" a deposition. E.g.:

- We create a deposit
- We poll its status 
- It reaches the max retries for polling, we mark it as `FAILED`
- Next day we want to sync the deposition status because it might actually have worked in SWH